sys/newlib/syscalls.c

```diff
@@ -85,15 +85,14 @@ void _exit(int n)
  *
  * @return [description]
  */
-caddr_t _sbrk_r(struct _reent *r, ptrdiff_t incr)
+void *_sbrk_r(struct _reent *r, ptrdiff_t incr)
 {
     unsigned int state = disableIRQ();
-    caddr_t res = heap_top;
+    void *res = heap_top;
 
-    if (((incr > 0) && ((heap_top + incr > &_eheap) || (heap_top + incr < res))) ||
-        ((incr < 0) && ((heap_top + incr < &_sheap) || (heap_top + incr > res)))) {
```

Comment:
It is only an issue when the heap is very large (more than 50% of the available address space) and the increment is large enough to cause an address overflow (positive or negative)

example:
```
&_sheap = 0x10000000
&_eheap = 0xff000000
heap_top = 0xf0000000
incr = 0x40000000
```
after addition: `heap_top = 0x30000000` the branch does not signal an error and instead of increasing the space, the space was reduced.